test/interactiveutil.jl

```diff
@@ -0,0 +1,117 @@
+### Testing download(url,[localfile],[silent=false])
+
+"Runs an expression, and returns stdout, stderr, and the result."
+function check_output(func::Function, args...; kwargs...)
+    originalSTDOUT = STDOUT
+    originalSTDERR = STDERR
+    (errRead, errWrite) = redirect_stderr()
+    (outRead, outWrite) = redirect_stdout()
+
+    try
+        result = func(args...; kwargs...)
+        # Ensure both streams are written to or readavailable waits forever.
+        write(STDOUT, "\n")
+        write(STDERR, "\n")
+
+        errstr = UTF8String(readavailable(errRead))
+        outstr = UTF8String(readavailable(outRead))
+        # Remove the manually added \n from stdout and stderr
+        return (chomp(outstr), chomp(errstr), result)
+    catch exception
+        # Immediately restore output so errors are printed.
+        redirect_stderr(originalSTDERR)
+        redirect_stdout(originalSTDOUT)
+        println("Error checking output of $func: args=$args, kwargs=$kwargs - $exception")
+    finally
+        redirect_stderr(originalSTDERR)
+        redirect_stdout(originalSTDOUT)
+        close(outRead)
+        close(errRead)
+        close(errWrite)
+        close(outWrite)
+    end
+end
+
+port = 2000 #Changes to next unused port after this
+html = "<html>success</html>"
+
+function start_server()
+    global server_running, port, html
```

Comment:
`html` doesn't need to be global. Maybe just pass it in as an argument to `start_server`?